src/ray/raylet/scheduling_policy.cc

```diff
@@ -36,22 +37,23 @@ std::unordered_map<TaskID, ClientID> SchedulingPolicy::Schedule(
                    << " numforwards=" << t.GetTaskExecutionSpec().NumForwards()
                    << " resources="
                    << t.GetTaskSpecification().GetRequiredResources().ToString();
```

Comment:
Just noticed that at [line 34](
https://github.com/ray-project/ray/blob/81cb95fc8dabc864e671e523b32c42927fa3fe04/src/ray/raylet/scheduling_policy.cc#L34) in this file, we have: `const auto &resource_demand = t.GetTaskSpecification().GetRequiredResources();`. I think it should be `const auto`.

Reply:
why? I'm using it read-only to check whether t.resreq <= node.avail - node.load. It's not mutated. Why do I need a copy?

Reply:
Ah never mind, I thought you would need it since `GetRequiredResources` returns a temporary variable, but seems the compiler will take care of it.